docs/topics/composite-primary-key.txt

```diff
@@ -185,3 +185,53 @@ field :exc:`.FieldError`.
     This is also true of composite primary keys. Hence, you may want to set
     :attr:`.Field.editable` to ``False`` on all primary key fields to exclude
     them from ModelForms.
+
+
+Building composite primary key ready applications
+=================================================
+
+Prior to the introduction of composite primary keys the single field composing
+the primary key of a model could be retrieved by introspecting the
+:attr:`primary key <django.db.models.Field.primary_key>` attribute of its
+fields:
+
+    .. code-block:: pycon
+
+        >>> pk_field = None
+        >>> for field in Product._meta.get_fields():
+        ...     if field.primary_key:
+        ...         pk_field = field
+        ...         break
+        ...
+        >>> pk_field
+        <django.db.models.fields.AutoField: id>
+
+Now that a primary key can be composed of multiple fields the
+:attr:`primary key <django.db.models.Field.primary_key>` attribute can no
+longer be relied upon to identify members of the primary key as it will be set
+to ``False`` to maintain the invariant that at most one field per model will
```

Comment:
nit: "at most one" implies it can be zero

Reply:
In the composite primary key case, there can be 0 fields with primary_key=True (which is in the example below I believe)

Reply:
I see your point

The way I always thought about this is it's still _exactly one_ field (the `pk` field)

This is why I was asking previously if we should force users to set `primary_key=True` on `CompositePrimaryKey` or not explicitly

Reply:
Ahh I see what you mean, I was considering `CompositePrimaryKey` to be a field itself given its _virtual_ nature.